stages/org.osbuild.ostree.deploy.container

```diff
@@ -105,10 +107,62 @@ def ostree_container_deploy(tree, inputs, osname, target_imgref, kopts):
             extra_args.append(f'--target-imgref={target_imgref}')
 
             kargs = [f'--karg={v}' for v in kopts]
-
             ostree.cli("container", "image", "deploy",
                        *extra_args, sysroot=tree, *kargs)
 
+            # the `ostree container image metadata` command utilizes the
+            # target-imgref, but only takes the name of the container image
+            target_imgref_list = target_imgref.split(':')
+            if target_imgref_list[0] == "ostree-remote-registry":
+                target_imgref_name = ':'.join(target_imgref_list[2:])
+            elif target_imgref_list[0] == ("ostree-image-signed" or "ostree-unverified-registry"):
+                target_imgref_name = ':'.join(target_imgref_list[1:])
+            
+            container_data = subprocess.run(["ostree", "container", "image", "metadata", f"--repo={tree}/ostree/repo", f"registry:{target_imgref_name}"],
+					check=True,
+					capture_output=True,
+					text=True).stdout.rstrip()
+            
+            container_data_config = subprocess.run(["ostree", "container", "image", "metadata", f"--repo={tree}/ostree/repo", f"registry:{target_imgref_name}", "--config"],
+					check=True,
+					capture_output=True,
+					text=True).stdout.rstrip()
+            
+            container_data_json = json.loads(container_data)
+            container_data_config_json = json.loads(container_data_config)
+
+            aleph_digest = container_data_json['config']['digest']
+            aleph_ref = f"docker://{target_imgref}"
+            aleph_ostree_version = container_data_config_json['config']['Labels']['org.opencontainers.image.version']
+            aleph_platform = "" # still up for discussion
+            aleph_imgid = "" # still up for discussion
+            aleph_container_image = container_data_config_json['config']['Labels']
+
+            aleph_version_data = {
+                "osbuild-version": osbuild.__version__,
+                "ref": aleph_ref,
+                "build": aleph_ostree_version,
+                "version": aleph_ostree_version,
+                "platform": aleph_platform,
+                "imgid": aleph_imgid,
+                "container-image": {
+                    "image-name": image_name,
+                    "image-digest": aleph_digest,
+                    "image-labels": aleph_container_image
+                }
+            }
+            
+            # the 'ostree.commit' label will be optional in the future so
+            # prevent hard failing if key is not found
+            if 'ostree.commit' in container_data_config_json['config']['Labels']:
+                aleph_ostree_commit = container_data_config_json['config']['Labels']['ostree.commit']
+                aleph_version_data["ostree-commit"] = aleph_ostree_commit
```

Comment:
@cgwalters To make `ostree-commit` optional, I add the key-value pair conditionally if the label exists. The only thing is that it appends it to the end of the dict. It's possible to insert the label between the `version` and `platform` labels where it was previously, but I don't think it's worth the trouble considering it will just make the code less readable and more confusing than it needs to be. Do you agree?

Reply:
I'd do this more like:
```
aleph_ostree_commit = container_data_config_json['config']['Labels'].get('ostree.commit')
if aleph_ostree_commit is not None:
    aleph_version_data["ostree-commit"] = aleph_ostree_commit
```

to avoid querying the dict twice.

As far as the ordering, well it's clearly fixable by splitting up the assignment to `aleph_verison_data` but I don't think anyone would care.  That said, it's actually probably a good idea to explicitly sort the output JSON anyways to aid reproducible builds.  Do this via passing `sort_keys=True` when invoking `json.dump`.